driver-core/src/test/unit/com/mongodb/internal/connection/X509AuthenticatorUnitTest.java

```diff
@@ -104,6 +107,18 @@ public void testSuccessfulAuthenticationAsync() throws ExecutionException, Inter
         validateMessages();
     }
 
+    @Test
+    public void testSpeculativeAuthentication() {
+        assumeTrue(serverVersionAtLeast(4, 3));
+
+        String speculativeAuthenticateResponse = "{\"dbname\": \"$external\", "
```

Comment:
Can we also assert that `createSpeculativeAuthenticateCommand` returns the expected document?